tests/test_send_telemetry_appinsights.py

```diff
@@ -76,3 +98,81 @@ def test_send_telemetry_appinsights_no_variant(self):
+        evaluation_event.reason = VariantAssignmentReason.DEFAULT_WHEN_ENABLED
+
+        with patch("featuremanagement.azuremonitor._send_telemetry.azure_monitor_track_event") as mock_track_event:
+            # This is called like this so we can override the track_event function
+            featuremanagement.azuremonitor._send_telemetry.publish_telemetry(  # pylint: disable=protected-access
+                evaluation_event
+            )
+            mock_track_event.assert_called_once()
+            assert mock_track_event.call_args[0][0] == "FeatureEvaluation"
+            assert mock_track_event.call_args[0][1]["FeatureName"] == "TestFeature"
+            assert mock_track_event.call_args[0][1]["Enabled"] == "True"
+            assert mock_track_event.call_args[0][1]["TargetingId"] == "test_user"
+            assert mock_track_event.call_args[0][1]["Variant"] == "big"
+            assert mock_track_event.call_args[0][1]["VariantAssignmentReason"] == "DefaultWhenEnabled"
+            assert mock_track_event.call_args[0][1]["VariantAssignmentPercentage"] == "50"
+            assert "DefaultWhenEnabled" in mock_track_event.call_args[0][1]
+            assert mock_track_event.call_args[0][1]["DefaultWhenEnabled"] == "big"
+
+    def test_send_telemetry_appinsights_allocation(self):
+        feature_flag = FeatureFlag.convert_from_json(
+            {
+                "id": "TestFeature",
+                "allocation": {
+                    "percentile": [{"from": 0, "to": 25, "variant": "big"}, {"from": 25, "to": 75, "variant": "small"}]
+                },
+            }
+        )
+        evaluation_event = EvaluationEvent(feature_flag)
+        variant = Variant("big", None)
+        evaluation_event.feature = feature_flag
+        evaluation_event.enabled = True
+        evaluation_event.user = "test_user"
+        evaluation_event.variant = variant
+        evaluation_event.reason = VariantAssignmentReason.PERCENTILE
+
+        with patch("featuremanagement.azuremonitor._send_telemetry.azure_monitor_track_event") as mock_track_event:
+            # This is called like this so we can override the track_event function
+            featuremanagement.azuremonitor._send_telemetry.publish_telemetry(  # pylint: disable=protected-access
+                evaluation_event
+            )
+            mock_track_event.assert_called_once()
+            assert mock_track_event.call_args[0][0] == "FeatureEvaluation"
+            assert mock_track_event.call_args[0][1]["FeatureName"] == "TestFeature"
+            assert mock_track_event.call_args[0][1]["Enabled"] == "True"
+            assert mock_track_event.call_args[0][1]["TargetingId"] == "test_user"
+            assert mock_track_event.call_args[0][1]["Variant"] == "big"
+            assert mock_track_event.call_args[0][1]["VariantAssignmentReason"] == "Percentile"
+            assert mock_track_event.call_args[0][1]["VariantAssignmentPercentage"] == "25"
```

Comment:
@rossgrambo this is 25, but if someone not in the 25 assigned percent is assigned the reason will be default when enabled but the variant assignment percentage will be 50.

Reply:
Yes that is correct and the weirdest part of what we landed on. 

`VariantAssignmentPercentage` shows the percentage of people expected to get this variant for the same `VariantAssignmentReason`. Most of the time `VariantAssignmentReason` will be percentile. When it's default- then the reason reflects the number of people who will get the default variant.